modules/core/src/smithy4s/http/RawErrorResponse.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright 2021-2024 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.http
+
+import scala.annotation.nowarn
+
+case class RawErrorResponse private (
+    private val codeField: Int,
+    private val headersField: Map[CaseInsensitive, Seq[String]],
+    private val bodyField: String,
+    private val failedDecodeAttemptField: FailedDecodeAttempt
+) extends Throwable {
+
+  def code: Int = codeField
+  def headers: Map[CaseInsensitive, Seq[String]] = headersField
+  def body: String = bodyField
+  def failedDecodeAttempt: FailedDecodeAttempt = failedDecodeAttemptField
```

Comment:
**question:** do we have to keep fields private and expose those getters? why not un-private fields in the case class def?

Reply:
Good question